packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -551,32 +562,38 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
             execute: (event?: Event) => {
                 const tabBar = this.findTabBar(event)!;
                 const currentIndex = tabBar.currentIndex;
-                this.shell.closeTabs(tabBar, (_, index) => index > currentIndex);
+                this.shell.closeTabs(tabBar, (title, index) => index > currentIndex && title.closable);
             }
         });
         commandRegistry.registerCommand(CommonCommands.CLOSE_ALL_TABS, {
-            isEnabled: (event?: Event) => this.findTabBar(event) !== undefined,
-            execute: (event?: Event) => this.shell.closeTabs(this.findTabArea(event)!)
+            isEnabled: (event?: Event) => {
+                const tabBar = this.findTabBar(event);
+                return tabBar !== undefined && tabBar.titles.some(title => title.closable);
+            },
+            execute: (event?: Event) => this.shell.closeTabs(this.findTabArea(event)!, title => title.closable)
         });
         commandRegistry.registerCommand(CommonCommands.CLOSE_MAIN_TAB, {
-            isEnabled: () => this.shell.getCurrentWidget('main') !== undefined,
+            isEnabled: () => {
+                const currentWidget = this.shell.getCurrentWidget('main');
+                return currentWidget !== undefined && currentWidget.title.closable;
+            },
             execute: () => this.shell.getCurrentWidget('main')!.close()
         });
         commandRegistry.registerCommand(CommonCommands.CLOSE_OTHER_MAIN_TABS, {
             isEnabled: () => {
                 const tabBars = this.shell.mainAreaTabBars;
-                return tabBars.length > 1 || tabBars.length === 1 && tabBars[0].titles.length > 1;
+                return tabBars.length > 1 || tabBars.length === 1 && tabBars[0].titles.some(title => title.closable);
```

Comment:
In case there are multiple tab bars, we should check the `closable` property, too. Maybe like this?
```typescript
const currentWidget = this.shell.getCurrentWidget('main');
return currentWidget !== undefined && this.shell.mainAreaTabBars.some(tb => tb.titles.some(title => title.widget !== currentWidget && title.closable));
```
Then you can safely remove the `currentWidget !== undefined` check in `execute`.

Reply:
Ok I was unsure here why in the first place it checked only one tabBar. I'll change it.